File: src/com/l2kt/gameserver/skills/conditions/ConditionPlayerRace.java
package com.l2kt.gameserver.skills.conditions;

import com.l2kt.gameserver.model.base.ClassRace;
import com.l2kt.gameserver.skills.Env;

/**
 * @author mkizub
 */
public class ConditionPlayerRace extends Condition
{
	private final ClassRace _race;
	
	public ConditionPlayerRace(ClassRace race)
	{
		_race = race;
	}
	
	@Override
	public boolean testImpl(Env env)
	{
		if (env.getPlayer() == null)
			return false;
		
		return env.getPlayer().getRace() == _race;
	}
}

File: src/com/l2kt/gameserver/skills/conditions/ConditionPlayerRace.kt
package com.l2kt.gameserver.skills.conditions

import com.l2kt.gameserver.model.base.ClassRace
import com.l2kt.gameserver.skills.Env

class ConditionPlayerRace(private val _race: ClassRace) : Condition() {

    override fun testImpl(env: Env): Boolean {
        return if (env.player == null) false else env.player!!.race == _race

    }
}